core/src/consensus.rs

```diff
@@ -112,26 +112,27 @@ impl Tower {
             let start_root = vote_state.root_slot;
 
             vote_state.process_slot_vote_unchecked(bank_slot);
-
-            for vote in &vote_state.votes {
-                Self::update_ancestor_lockouts(&mut stake_lockouts, &vote, ancestors);
-            }
+            let mut votes = vec![];
             if start_root != vote_state.root_slot {
```

Comment:
@carllin not sure if roots are necessary

Reply:
They probably don't make too big of a difference as they're implemented now. I was thinking we maybe keep track of the latest roots for each validator, and only include the weight of the root for a vote state if it's the latest known root for that validator